crypto/txscript/src/opcodes/macros.rs

```diff
@@ -121,5 +121,43 @@ macro_rules! opcode_list {
                 _ => None
             }
         }
+
+        #[cfg(test)]
+        use crate::script_builder::{ScriptBuilder, ScriptBuilderResult};
+
+        #[cfg(test)]
+        #[allow(unused_comparisons)]
+        pub(crate) fn parse_short_form(test: String) -> ScriptBuilderResult<Vec<u8>>
+        {
+            let mut builder = ScriptBuilder::new();
+            for token in test.split_whitespace() {
+                if let Ok(value) = token.parse::<i64>() {
+                    builder.add_i64(value)?;
+                }
+                else if let Ok(value) = token.strip_prefix("0x").and_then(|trimmed| Some(hex::decode(trimmed))).unwrap_or_else(|| Err(hex::FromHexError::InvalidStringLength)) {
```

Comment:
Not sure what's the use of `unwrap_or_else` in here

Reply:
I think this is a better alternative if I'm understanding correctly what you're trying to achieve

```
  else if let Some(Ok(value)) = token.strip_prefix("0x")
      .and_then(|trimmed| Some(hex::decode(trimmed))) {
      builder.extend(&value);
  }
```

Although in test context it's safe to assume hex is valid, so this will also do the job:

```
else if let Some(value) = token.strip_prefix("0x")
                    .and_then(|trimmed| Some(hex::decode(trimmed).unwrap())) {
                    builder.extend(&value);
}
```

Reply:
Changed to the first option.